# Upstream status: approved by the fb sub-system maintainer.  I hope he'll
# send it on to Linus or 2.6.17.

Date: Tue, 9 May 2006 23:01:14 +0200
From: Martin Michlmayr <tbm@cyrius.com>
To: linux-fbdev-devel@lists.sourceforge.net
Subject: [PATCH] Fix maxine compilation error

[PATCH] Fix maxine compilation error

Fix the following compilation error:

  CC      drivers/video/maxinefb.o
drivers/video/maxinefb.c:58: warning: initializer-string for array of chars is too long
drivers/video/maxinefb.c:58: warning: (near initialization for ‘maxinefb_fix.id’)
drivers/video/maxinefb.c:110: error: unknown field ‘fb_get_fix’ specified in initializer
drivers/video/maxinefb.c:110: error: ‘gen_get_fix’ undeclared here (not in a function)
drivers/video/maxinefb.c:111: error: unknown field ‘fb_get_var’ specified in initializer
drivers/video/maxinefb.c:111: error: ‘gen_get_var’ undeclared here (not in a function)
make[2]: *** [drivers/video/maxinefb.o] Error 1

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>


--- a/drivers/video/maxinefb.c~	2006-05-09 20:50:36.000000000 +0000
+++ b/drivers/video/maxinefb.c	2006-05-09 20:57:27.000000000 +0000
@@ -55,7 +55,7 @@
 };
 
 static struct fb_fix_screeninfo maxinefb_fix = {
-	.id =		"Maxine onboard graphics 1024x768x8",
+	.id =		"Maxine",
 	.smem_len =	(1024*768),
 	.type =		FB_TYPE_PACKED_PIXELS,
 	.visual =	FB_VISUAL_PSEUDOCOLOR,
@@ -107,8 +107,6 @@
 
 static struct fb_ops maxinefb_ops = {
 	.owner		= THIS_MODULE,
-	.fb_get_fix	= gen_get_fix,
-	.fb_get_var	= gen_get_var,
 	.fb_setcolreg	= maxinefb_setcolreg,
 	.fb_fillrect	= cfb_fillrect,
 	.fb_copyarea	= cfb_copyarea,

